test/controllers/hosts_controller_test.rb

```diff
@@ -1263,38 +1263,47 @@ class Host::Valid < Host::Managed; end
   end
 
-  test 'template_used returns templates with interfaces' do
-    @host.setBuild
-    nic=FactoryGirl.create(:nic_managed, :host => @host)
-    attrs = host_attributes(@host)
-    attrs[:interfaces_attributes] = nic.attributes.except 'updated_at', 'created_at', 'attrs'
-    ActiveRecord::Base.any_instance.expects(:destroy).never
-    ActiveRecord::Base.any_instance.expects(:save).never
-    xhr :put, :template_used, {:provisioning => 'build', :host => attrs, :id => @host.id }, set_session_user
-    assert_response :success
-    assert_template :partial => '_provisioning'
-  end
+  describe '#template_used' do
+    setup do
+      @host.setBuild
+      ActiveRecord::Base.any_instance.expects(:destroy).never
+      ActiveRecord::Base.any_instance.expects(:save).never
+      @attrs = host_attributes(@host)
+    end
 
-  test 'template_used returns templates with host parameters' do
-    @host.setBuild
-    attrs = host_attributes(@host)
-    attrs[:host_parameters_attributes] = {'0' => {:name => 'foo', :value => 'bar', :id => '34'}}
-    ActiveRecord::Base.any_instance.expects(:destroy).never
-    ActiveRecord::Base.any_instance.expects(:save).never
-    xhr :put, :template_used, {:provisioning => 'build', :host => attrs }, set_session_user
-    assert_response :success
-    assert_template :partial => '_provisioning'
-  end
+    test 'returns templates with interfaces' do
+      nic=FactoryGirl.create(:nic_managed, :host => @host)
+      @attrs[:interfaces_attributes] = nic.attributes.except 'updated_at', 'created_at', 'attrs'
+      xhr :put, :template_used, {:provisioning => 'build', :host => @attrs, :id => @host.id }, set_session_user
+      assert_response :success
+      assert_template :partial => '_provisioning'
+    end
 
-  test 'template_used does not save has_many relations on existing hosts' do
-    @host.setBuild
-    attrs = host_attributes(@host)
-    attrs[:config_group_ids] = [config_groups(:one).id]
-    ActiveRecord::Base.any_instance.expects(:destroy).never
-    ActiveRecord::Base.any_instance.expects(:save).never
-    xhr :put, :template_used, {:provisioning => 'build', :host => attrs, :id => @host.id }, set_session_user
-    assert_response :success
-    assert_template :partial => '_provisioning'
+    test 'returns templates with host parameters' do
+      @attrs[:host_parameters_attributes] = {'0' => {:name => 'foo', :value => 'bar', :id => '34'}}
+      xhr :put, :template_used, {:provisioning => 'build', :host => @attrs }, set_session_user
+      assert_response :success
+      assert_template :partial => '_provisioning'
+    end
+
+    test 'does not save has_many relations on existing hosts' do
+      @attrs[:config_group_ids] = [config_groups(:one).id]
+      xhr :put, :template_used, {:provisioning => 'build', :host => @attrs, :id => @host.id }, set_session_user
+      assert_response :success
+      assert_template :partial => '_provisioning'
+    end
+
+    test 'shows templates for image provisioning' do
```

Comment:
This is the only bit that was added. The rest is merely refactoring a few lines of each `template_used` related test into a `setup` block.